book/src/usage.md

```diff
@@ -37,19 +37,30 @@ If a register is selected before invoking a change or delete command, the select
 - `"hc` - Store the selection in register `h` and then change it (delete and enter insert mode).
 - `"md` - Store the selection in register `m` and delete it.
 
-### Special registers
+### Default registers
+
+Commands that use registers, like yank (`y`), use a default register if none is specified.
+These registers are used as defaults:
 
 | Register character | Contains              |
 | ---                | ---                   |
 | `/`                | Last search           |
 | `:`                | Last executed command |
 | `"`                | Last yanked text      |
-| `_`                | Black hole            |
+| `@`                | Last recorded macro   |
+
+### Special registers
 
-The system clipboard is not directly supported by a special register. Instead, special commands and keybindings are provided. Refer to the
-[key map](keymap.md#space-mode) for more details.
+Some registers have special behavior when read from or written to:
```

Comment:
Yeah this distinction is cleaner in Kakoune because all special registers are effectively read-only (though the blackhole is meant to be written to). This PR uses `write`/`push` for the clipboard registers. I like the idea of two more columns for reading and writing behaviors 👍

Reply:
Hmm actually I think it could make sense to separate out clipboard registers from special registers in the docs since all of the special registers are read-only

Reply:
You can write to the blackhole register tough but we could just put that with the system clipboard and make one read-only and one read-write table.

Reply:
In Kakoune at least you can write to all of the special registers and they silently discard the values like the blackhole register. We could return an Err though with a message that they're unwritable. What do you think?

Reply:
I dont feel to strongly either way but to me it seems better to return an error here. If I want to dicard, I can use the black hole register.

All other registers are really not intended for writing and allowing it is more of a "why not". It kind of makes the black hole register a bit pointless too (you could just use `#` for the same thing mostly). So returning an error feels more consistent to me. 

If somebody used one of these special register for writing it would be more likely a typo instead of intentional (since then they would just use `_`) so an error feels like.better ui.

Reply:
Ok I made this refactor. Lmk what you think about the defaults for `write` and `push` in this case. (Is there a nicer way to express this with a ReadOnlyRegister` trait of some sort?)

Reply:
you could in theory add a second `ReadOnlyRegister` trait and then add an automatic register implementation that just forwards `Register::read` to `ReadOnlyRegister::read` and returns an error on `write`.

``` rust
impl<T: ReadOnlyRegister> Register for T{
 ...
}
```


The default implementations are a much simpler solution tough and the extra boilerplate might not be worth the extra clarity. I think the default implementations make a lot of sense :+1: